### PR TITLE
[Aurora Vulnerability Fix] Add Rate Limiting to all requests for server routes

### PR DESCRIPTION
**Pull request by HarryZ10**
## Description of the change

> This PR adds a little more security, adding a rate limit on all requests e.g `/`. In effect of this PR, a new `express-rate-limit` package will be added to the project. The configurations for this rate limit call are 100 requests per 15-minute window.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes [#801](https://github.com/Recidiviz/zenhub-tasks/issues/801)

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by phenggeler** [comment]
@HarryZ10 - could you add @Catacola and @danawillow  as reviewers as well? Thanks!

**Comment by lilidworkin** [comment]
FWIW, both Case Triage and Justice Counts have some basic rate limiting in place, but I'm not sure if we do it the right way or if it's configured properly! Just wanted to pass this along in case this is part of a bigger project to get rate limiting implemented everywhere. [Here's](https://github.com/Recidiviz/recidiviz-data/blob/a30721ed8108f32d7f2bba71623c211a37496697/recidiviz/justice_counts/control_panel/server.py#L107) where we have it in justice counts and [here's](https://github.com/Recidiviz/recidiviz-data/blob/a30721ed8108f32d7f2bba71623c211a37496697/recidiviz/case_triage/server.py#L96) where it is in case triage.

**Comment by phenggeler** [comment]
> FWIW, both Case Triage and Justice Counts have some basic rate limiting in place, but I'm not sure if we do it the right way or if it's configured properly! Just wanted to pass this along in case this is part of a bigger project to get rate limiting implemented everywhere. [Here's](https://github.com/Recidiviz/recidiviz-data/blob/a30721ed8108f32d7f2bba71623c211a37496697/recidiviz/justice_counts/control_panel/server.py#L107) where we have it in justice counts and [here's](https://github.com/Recidiviz/recidiviz-data/blob/a30721ed8108f32d7f2bba71623c211a37496697/recidiviz/case_triage/server.py#L96) where it is in case triage.

Thanks @lilidworkin - this came out of an effort to tackle CodeQL security issues. @HarryZ10 - I'd defer to the method @lilidworkin and other teams are already using

File: server/app.js [hunk 1]
```diff
@@ -38,6 +39,13 @@ const app = express();
 
 const upload = multer();
 
+const limiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 100, // each IP address gets 100 requests per 15 minutes
```
[the remark this answers is not in the document]
Comment (HarryZ10) [reply server/app.js]: Fixed! @danawillow @lilidworkin 

Thanks for the catch!

**Review by danawillow — APPROVED** [review]

**Pull request merged**